### PR TITLE
Implementation of Table for Android

### PR DESCRIPTION
**Pull request by t-arn**
This PR adds the implementation of the Table widget for Android

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

Screenshot_1620797059

### CODE REVIEW

**Comment by t-arn** [comment]
Here is how the example looks like on Windows:
image

And this is on Android:
image
Why are the labels 'Ready' and 'Try multiple row selection' not at the bottom of the screen?

**Comment by paulproteus** [comment]
Hi @t-arn ! This is great to see. The labels aren't visible I think due to a layout bug I introduced :( where the vertical space used by "Table Demo" (app header) isn't subtracted properly from what we pass to Toga layout. I would love help figuring this out, but hope to spend some time on it eventually.

**Comment by t-arn** [comment]
Hi @paulproteus, the labels **are** there, but they are right below the table and not at the bottom of the window

**Comment by t-arn** [comment]
@paulproteus I found and fixed the problem of the labels not being at the bottom of the window (see commit 4a7aa90).
I took the (incorrectly working) code from your DetailedList implementation. 
Are we having the same problem (with flex=1 not using the vertical space) also in DetailedList?

File: examples/table/table/app.py [hunk 1]
```diff
@@ -72,7 +81,7 @@ def startup(self):
         self.table1 = toga.Table(
             headings=headings,
             data=bee_movies[:4],
-            style=Pack(flex=1, padding_right=5),
+            style=Pack(flex=1, padding_right=5, height=250),
```
Comment (freakboy3742) [inline examples/table/table/app.py:84]: This height modifier shouldn't be required; the fact that it is required is a red flag.

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
This looks pretty good as a first pass. I've flagged one thing inline about the change to layout in the demo code; three other general notes:

1) As you've noted, the "N rows selected" footer isn't visible; I would have thought this was addressed by the @paulproteus's recent fix to overall height calculation, but it doesn't appear to be.
2) The header row isn't visually distinguished from the table content - they look like they're "just another row of data". 
3) The header row scrolls with the content, rather than being a true "header". In every other platform, the header row is part of the "container", rather than being data; so even if you've got multiple pages of content, you can always see the header. 

If you can give examples of how (2) and (3) are idiomatic (i.e., that's how tables are *meant* to work) on Android, then that's fine - but it seems like an odd design choice as is.

File: examples/table/table/app.py (hunk 1 above)
Comment (t-arn) [reply examples/table/table/app.py:84]: @freakboy3742 Thank you for your review! 
I think (1) should be solved - I just did not yet apply Asheesh's fix to this branch. 
As for (2), I searched the Internet but did not find any recommendation about table header styles in Android. In Material Design, they recommend using a bold font with no special background color. Would that be acceptable for you?
As for (3), from what I found and know, fixed headers are not the normal way to create a table in Android. I found some posts where they tried hard to implement a fixed header row and it involved using 2 nested tables.
@paulproteus What is your opinion regarding the points (2) and (3) ?

**Comment by t-arn** [comment]
@freakboy3742 Your issue (1) is now solved. To address (2), the table headers are now in bold like in Material Design.

**Comment by t-arn** [comment]
As for (3) I believe that this is the way tables are meant to work on Android. I found several posts from people who wanted to implement fixed table headers, but they did not find a clean way to implement this. They usually ended up with  creating 2 tables - one for the headers and one for the data.

**Review by freakboy3742 — APPROVED** [review]
Awesome - thanks for those updates and clarifications.

There's still some odd behavior with the scrollbar (e.g., the horizontal scrollbar is always at the bottom of the visible content, rather than the bottom of the visible *area*), but that's consistent with the odd behaviour we've seen in ScrollView, so I don't think it needs to be a blocker on merging this.

I've made a couple of minor cleanups, but otherwise, this is good to go! Thanks for all the hard work!

**Pull request merged**